### PR TITLE
Initial website optimization & first post added (see: https://github.…

### PR DESCRIPTION
**Pull request by kamilpytlak**
https://github.com/ttscience/ttscience_blog/issues/2

### CODE REVIEW

File: config.yaml [hunk 1]
```diff
@@ -13,20 +13,57 @@ services:
...
+  ShowCodeCopyButtons: true
   homeInfoParams:
-    Title: "Hi there 👋"
-    Content: "Welcome to the TT Science Development Blog! Here, you'll find insights, tutorials, and updates about our work in clinical data science, including R, Python, and optimization techniques. Explore our posts, check out our projects on GitHub, or learn more about our team."
+    Title: "Hey there, data enthusiasts! 👋"
+    Content: "Here at TT Science, we are a dynamic team of individuals — like trees in a [random forest](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.RandomForestRegressor.html) 🌳🌲 — each bringing our own unique strengths and perspectives to the table. While we share a common goal of advancing clinical data science, it’s our diversity that makes us truly special! <br><br> 🌈 Join us on this exciting journey 🚀 as we share insights, tutorials, and updates on our innovative projects involving R, Python, and biostatistics. 💻🔍 Explore our posts, dive into our GitHub projects 📂, and get to know the passionate minds behind our work. Together, we’re harnessing the power of data to create impactful solutions in healthcare. 💡💖 Welcome to our forest of knowledge! 🌳🌟"
```
Comment (salatak) [inline config.yaml:45]: The intro sounds pretty good, but we're overusing emoticons. It makes it look a bit childish and overwhelming at the same time. I think it's better to aim for a 'casual elegant' tone in writing. I like the second part of the intro the most, and I'd consider tweaking the first part, although it's not essential.

File: config.yaml [hunk 2]
```diff
@@ -13,20 +13,57 @@ services:
...
+  assets:
+    favicon: "favicon.ico"
+    favicon16x16: "favicon-16x16.png"
+    favicon32x32: "favicon-32x32.png"
+    apple_touch_icon: "apple-touch-icon.png"
+  socialIcons:
+    - name: "Company website"
+      url: "https://ttsi.com.pl/"
```
Comment (salatak) [inline config.yaml:53]: Is it possible to use the TTSI logo as an icon for linking to our website? I think that would be clear what we are clicking on.

File: content/about.md [hunk 3]
```diff
@@ -1,6 +1,17 @@
...
+
+We believe in science and want innovative scientists to tackle the questions and challenges of today’s world. We want scientists to always be able to rely on data—because we all rely on them. Our partnership approach is rooted in the scientific method—we combine data, our clients’ expertise, and our competencies to achieve the best results. 
+
+Our mission is to share knowledge with the broader community engaged in medical IT software development, clinical research, and the analysis of medical and biological data, as well as those just beginning their careers in these fields. Through our published content, we aim to showcase solutions for software development and provide guidance to researchers, particularly in the areas of statistics, machine learning, and the regulatory frameworks of the European Medicines Agency (EMA) and the U.S. Food and Drug Administration (FDA).
+
+If you want to check our official page, visit Transition Technologies Science—you’ll learn that we provide software, data science services, and offer analytical support at every stage of clinical trials.
+
+Reach out to us to explore the world of data - our email is office@ttsi.com.pl
```
Comment (salatak) [inline content/about.md:17]: The `about me` section is usually a bit more concise. Maybe we could shorten it and add some photos from conferences or our team-building events? I think if people see a wall of text, they might get discouraged. We don’t want an overly serious blog, but rather a professional one with a relaxed tone.

Comment (salatak) [reply content/about.md:17]: The text itself is good, but I would place the third and fourth paragraphs as a welcome post that outlines our intentions. In the "about me" section, I would keep the focus on us and our work, rather than what’s on the blog.

File: content/post/2024-10-01-same-same-but-different-how-advanced-data-science-techniques-help-us-validate-drug-names/index.md [hunk 4]
```diff
@@ -0,0 +1,623 @@
...
+
+
+``` r
+set.seed(7)
+drug_names <- atc_df |>
+  filter(nchar(atc_code) == 7) |>
+  slice_sample(n = 100, replace = TRUE) |>
+  pull(atc_name)
+
+unique(drug_names) |> sort()
```
Comment (salatak) [inline content/post/2024-10-01-same-same-but-different-how-advanced-data-science-techniques-help-us-validate-drug-names/index.md]: When writing code, let’s try to demonstrate good practices in R from the very beginning. When using functions from a specific library, we should specify the library right before the function so it’s clear for the reader where to look for the function later. For example, use `dplyr::mutate()` instead of just `mutate()`. Some functions exist in different libraries with the same name, so it's important to always indicate the library from which they originate.

**Review by salatak — CHANGES_REQUESTED** [review]
I like the way the user is guided through the topic, but I would expect a clearer indication of the benefits that come from using the proposed solution.

File: config.yaml (hunk 2 above)
Comment (kamilpytlak) [reply config.yaml:53]: Sure, updated

File: content/post/2024-10-01-same-same-but-different-how-advanced-data-science-techniques-help-us-validate-drug-names/index.md (hunk 4 above)
Comment (kamilpytlak) [reply content/post/2024-10-01-same-same-but-different-how-advanced-data-science-techniques-help-us-validate-drug-names/index.md]: Changed.

File: content/about.md (hunk 3 above)
Comment (kamilpytlak) [reply content/about.md:17]: @aleksandraduda2

File: config.yaml (hunk 2 above)
Comment (kamilpytlak) [reply config.yaml:53]: https://github.com/ttscience/ttscience_blog/pull/6/commits/e41dc112f2d0e3213425947501b1bcb6a5e80ecb

File: content/post/2024-10-01-same-same-but-different-how-advanced-data-science-techniques-help-us-validate-drug-names/index.md (hunk 4 above)
Comment (kamilpytlak) [reply content/post/2024-10-01-same-same-but-different-how-advanced-data-science-techniques-help-us-validate-drug-names/index.md]: https://github.com/ttscience/ttscience_blog/pull/6/commits/e41dc112f2d0e3213425947501b1bcb6a5e80ecb

File: config.yaml (hunk 1 above)
Comment (kamilsi) [reply config.yaml:45]: Also we have already "about us" + "authors" - plenty of places where we talk "about us". My proposition: let's revert to old version where this was just a friendly waving hand.

File: content/about.md (hunk 3 above)
Comment (kamilsi) [reply content/about.md:17]: I would leave main page empty, without "about us" or just one sentence. I agree with shortening - removing 4th paragraph is fine.

File: .Rprofile [hunk 5]
```diff
@@ -1,3 +1,4 @@
+source("renv/activate.R")
```
Comment (kamilsi) [inline .Rprofile:1]: I cannot work on my laptop already - some C compilation issues :-/ Probably renv was a good idea, but it's going to be complicated.

File: assets/css/common/post-single.css [hunk 6]
```diff
@@ -34,23 +34,27 @@
 
 .post-content {
     color: var(--content);
+    text-align: justify;
```
Comment (kamilsi) [inline assets/css/common/post-single.css:37]: I wouldn't say I like justified text aesthetically (especially on webpages, docs are OK). @salatak what do you think?

File: content/post/2024-10-01-same-same-but-different-how-advanced-data-science-techniques-help-us-validate-drug-names/index.md [hunk 7]
```diff
@@ -0,0 +1,628 @@
...
+  - statistics
+  - text analysis
+tags:
+  - data visualization
+  - drug names
+  - eCRF
+  - data validation
+  - levenshtein distance
+  - NLP
+  - t-SNE
```
Comment (kamilsi) [inline content/post/2024-10-01-same-same-but-different-how-advanced-data-science-techniques-help-us-validate-drug-names/index.md:18]: What is the difference between tags and categories? I think only one of these is presented.

File: content/post/2024-10-01-same-same-but-different-how-advanced-data-science-techniques-help-us-validate-drug-names/index.md [hunk 8]
```diff
@@ -0,0 +1,628 @@
...
+slug: same-same-but-different-how-advanced-data-science-techniques-help-us-validate-drug-names
+ShowToc: yes
```
Comment (kamilsi) [inline content/post/2024-10-01-same-same-but-different-how-advanced-data-science-techniques-help-us-validate-drug-names/index.md:20]: It's a very long post if you need a ToC :-)

File: content/post/2024-10-01-same-same-but-different-how-advanced-data-science-techniques-help-us-validate-drug-names/index.md [hunk 9]
```diff
@@ -0,0 +1,628 @@
...
+
+# Randomly select medications (including those with typos)
+con_meds <- sample(medications_typo, length(patient_ids), replace = TRUE)
+
+con_meds_df <- data.frame(
+  patient_id = patient_ids,
+  con_med = medications_typo
+)
+
+print(con_meds_df)
```
Comment (kamilsi) [inline content/post/2024-10-01-same-same-but-different-how-advanced-data-science-techniques-help-us-validate-drug-names/index.md:52]: print heads, printing everything doesn't add value

File: content/post/2024-10-01-same-same-but-different-how-advanced-data-science-techniques-help-us-validate-drug-names/index.md [hunk 10]
```diff
@@ -0,0 +1,628 @@
...
+
+
+## Clustering - That Is, Group Similar Into Similar
+
+The graphics can be tricky to interpret, especially when we have a lot of drugs. That's why we use an algorithmic approach based on clustering "similar into similar" in our workflow. One clustering technique we find really effective is DBSCAN (Density-Based Spatial Clustering of Applications with Noise). This algorithm is great at identifying clusters of different shapes and sizes, which helps us find meaningful patterns in the complex landscape of drug names. By looking at how dense the data is, DBSCAN groups similar drug names together while also showing us any outliers that might be typos. For example, it can group entries like "Ibuprofin" and "Ibuprofine," showing us any potential inconsistencies that could affect clinical trial results. This powerful approach not only makes the data more reliable but also makes it easier to clean up, so researchers can focus on getting good results in their clinical studies.
+
+
+# Case Study: Validation of Drug Names for a Pharmaceutical Gig
```
Comment (kamilsi) [inline content/post/2024-10-01-same-same-but-different-how-advanced-data-science-techniques-help-us-validate-drug-names/index.md:109]: I was wondering if we might want to consider either simplifying headers or possibly removing them altogether. Since we’re aiming for a ‘serious friend’ tone, it might work better to let the content speak for itself without too much emphasis on sub-headers. What do you think?

File: content/post/2024-10-01-same-same-but-different-how-advanced-data-science-techniques-help-us-validate-drug-names/index.md [hunk 11]
```diff
@@ -0,0 +1,628 @@
...
+
+Have you already figured out what problems this can cause? Computers are pretty simple — to them, "aspirin" and "Aspirin" are two different text strings. So, when we're counting the proportions of patients taking different drugs at different stages of treatment, we'll get two separate proportions for aspirin and Aspirin, even though they're the same drugs.
+
+So what solutions do we have?
+- See what unique drugs [`unique(con_meds)`] our database contains, **find errors and just correct them**.
+- See what unique drugs [`unique(con_meds)`] our database contains, **calculate the distance between their names, and then visualize them**.
+
+If our database has a relatively small number of con-meds, the first option is probably the easiest. However, the more drugs patients are taking, the more complex it becomes. This is because different drugs can have similar names, and sometimes patients have different characteristics that affect their drug regimen. To quickly and effectively identify duplicates, ambiguities, and errors in drug names, we need a more efficient method. **This is where Levenshtein  and t-SNE come in handy.**
```
Comment (kamilsi) [inline content/post/2024-10-01-same-same-but-different-how-advanced-data-science-techniques-help-us-validate-drug-names/index.md:125]: A lot of the content here seems to repeat points already covered earlier in the article. For example:
- The challenges with inconsistent drug names (e.g., “aspirin” vs. “Aspirin”) are already explained in detail in the ‘Introduction’ section.
- The concept of using distance metrics like Levenshtein and visualization methods like t-SNE is also well-covered in the sections on ‘t-SNE + Distance Metrics’ and ‘Clustering’.

Because these ideas have already been introduced and explained clearly, this section feels redundant. Streamlining by removing it could help keep the reader focused and improve the article’s flow. What do you think?

File: content/post/2024-10-01-same-same-but-different-how-advanced-data-science-techniques-help-us-validate-drug-names/index.md [hunk 12]
```diff
@@ -0,0 +1,628 @@
...
+##  [92] "ticarcillin"                                   
+##  [93] "tiemonium iodide and analgesics"               
+##  [94] "timolol, thiazides and other diuretics"        
+##  [95] "tolperisone"                                   
+##  [96] "tramadol"                                      
+##  [97] "tretoquinol"                                   
+##  [98] "trypsin, combinations"                         
+##  [99] "ursodoxicoltaurine"                            
+## [100] "zidovudine"
+```
```
Comment (kamilsi) [inline content/post/2024-10-01-same-same-but-different-how-advanced-data-science-techniques-help-us-validate-drug-names/index.md:281]: Feels a bit redundant, as the structure of the codebook has already been presented earlier. Listing all the names in full doesn’t seem to add much new information and might distract the reader.

File: content/post/2024-10-01-same-same-but-different-how-advanced-data-science-techniques-help-us-validate-drug-names/index.md [hunk 13]
```diff
@@ -0,0 +1,628 @@
...
+    # Swap two adjacent characters
+    swap_pos <- sample(1:(length(name_chars) - 1), 1)
+    temp <- name_chars[swap_pos]
+    name_chars[swap_pos] <- name_chars[swap_pos + 1]
+    name_chars[swap_pos + 1] <- temp
+  }
+    
+  return(paste(name_chars, collapse = ""))
+}
+```
```
Comment (kamilsi) [inline content/post/2024-10-01-same-same-but-different-how-advanced-data-science-techniques-help-us-validate-drug-names/index.md:310]: The introduce_variation function adds complexity that might not be necessary for this article. Since the focus is on fixing errors in drug names, introducing random typos as part of the workflow could confuse readers. Pre-preparing a small set of intentional errors and using them consistently would simplify the explanation and keep the focus on error correction rather than error creation.

File: content/post/2024-10-01-same-same-but-different-how-advanced-data-science-techniques-help-us-validate-drug-names/index.md [hunk 14]
```diff
@@ -0,0 +1,628 @@
...
+complete_drug_names <- c(drug_names, sample_drug_names_with_typos)
+
+# Add additional drug names in Polish
+complete_drug_names <- c(complete_drug_names, c("Kwas acetylosalicylowy i kortykosteroidy", # Acetylsalicylic acid and corticosteroids
+                                                "Węglan magnezu", # Magnesium carbonate
+                                                "Kwas foliowy" # Folic acid
+))
+
+unique(complete_drug_names) |> sort() |> head(10)
+```
```
Comment (kamilsi) [inline content/post/2024-10-01-same-same-but-different-how-advanced-data-science-techniques-help-us-validate-drug-names/index.md:335]: Polish is confusing

File: content/post/2024-10-01-same-same-but-different-how-advanced-data-science-techniques-help-us-validate-drug-names/index.md [hunk 15]
```diff
@@ -0,0 +1,628 @@
...
+## Perplexity: 3 | KL Divergence: 1.467266 
+## Best Perplexity So Far: 2 | Best KL Divergence: 1.365567 
+## 
+## Perplexity: 4 | KL Divergence: 1.605605 
+## Best Perplexity So Far: 2 | Best KL Divergence: 1.365567 
+## 
+## Perplexity: 5 | KL Divergence: 1.612329 
+## Best Perplexity So Far: 2 | Best KL Divergence: 1.365567 
+## 
+## Perplexity: 6 | KL Divergence: 1.841238
```
Comment (kamilsi) [inline content/post/2024-10-01-same-same-but-different-how-advanced-data-science-techniques-help-us-validate-drug-names/index.md:456]: I would skip this, doesn't add much

File: content/post/2024-10-01-same-same-but-different-how-advanced-data-science-techniques-help-us-validate-drug-names/index.md [hunk 16]
```diff
@@ -0,0 +1,628 @@
...
+tsne_df |>
+  ggplot(aes(x = Dim1, y = Dim2, label = Label)) +
+  geom_point(aes(color = Cluster), size = 1) +
+  geom_text_repel(aes(label = Label), max.overlaps = 10, size = 2) +
+  labs(x = "Dimension 1", y = "Dimension 2") +
+  guides(color = "none") +
+  theme_bw()
+```
+
+<img src="{{< blogdown/postref >}}index_files/figure-html/unnamed-chunk-11-1.png" width="672" />
```
Comment (kamilsi) [inline content/post/2024-10-01-same-same-but-different-how-advanced-data-science-techniques-help-us-validate-drug-names/index.md:512]: better for SEO to name chunks

**Review by kamilsi — COMMENTED** [review]
Great work on this article—it’s thorough and informative! I have a few suggestions to refine it further:

1. **Tone:** Since we’re aiming for a 'serious friend' tone, I’d recommend avoiding casual language, such as in some headers (e.g., 'Pharmaceutical Gig'). Keeping a professional yet approachable tone would better suit our goals.

2. **Focus:** To keep the article sharp and engaging, I’d suggest cutting out anything that isn’t directly about fixing typos in drug names. For example, the `introduce_variation` function for generating typos feels tangential and might distract readers from the main topic.

3. **Streamlining:** There are a few opportunities to streamline the content:
   - Avoid repetitions of ideas already covered, like the challenges with inconsistent drug names.
   - Minimize code output, such as printing entire data frames or diagnostics, which can overwhelm readers. Summarizing key outputs or providing representative samples instead would keep the focus tight and improve readability.

These changes could make the article clearer, more concise, and more aligned with its purpose. Let me know what you think!

File: assets/css/common/post-single.css (hunk 6 above)
Comment (kamilpytlak) [reply assets/css/common/post-single.css:37]: Comparison:

image

vs.

image

Please decide, it's all the same to me.

File: config.yaml (hunk 1 above)
Comment (kamilpytlak) [reply config.yaml:45]: OK

Comment (kamilpytlak) [reply config.yaml:45]: What do you think of such a variation (without “Content” is left a little blank)?: 
image

File: content/about.md (hunk 3 above)
Comment (kamilpytlak) [reply content/about.md:17]: Resolved.
